table/block_based/block_based_table_iterator.cc

```diff
@@ -557,5 +700,119 @@ void BlockBasedTableIterator::FindReadAheadSizeUpperBound() {
 
   block_prefetcher_.SetUpperBoundOffset(start_offset +
                                         total_bytes_till_upper_bound);
+
+  if (getenv("Print")) {
+    printf("FindReadAheadSizeUpperBound:End\n");
+  }
+}
+
+void BlockBasedTableIterator::BlockCacheLookupForReadAheadSize(
```

Comment:
Should we pass in the offset as well? Atleast we can validate the assumption that current block handle + size should be at the offset.

Also, will the readahead_size here be size of missing block + size of prefetch, or just the prefetch size?

Reply:
1. Sure! I'll pass the offset

2. readahead_size is of just the size of prefetch. Because in FilePrefetchBuffer, both are treated separately. Its always prefetch (missing block + readahead_size). So keeping the behavior here same.

Reply:
Done!